mirrord/config/src/feature/env.rs

```diff
@@ -111,6 +113,26 @@ pub struct EnvConfig {
     /// These variables will override environment fetched from the remote target.
     #[config(env = MIRRORD_OVERRIDE_ENV_FILE_ENV)]
     pub env_file: Option<PathBuf>,
+
+    /// ### feature.env.mapping {#feature-env-mapping}
+    ///
+    /// Specify map of patterns that if matched will replace the value according to specification.
+    ///
+    /// *Capture groups are allowed.*
+    ///
+    /// Example:
+    /// ```json
+    /// {
+    ///   ".+_TIMEOUT": "10000"
+    ///   "LOG_.+_VERBOSITY": "debug"
+    /// }
+    /// ```
+    ///
+    /// Will do the next replacements for environment variables that match:
+    ///
+    /// `CONNECTION_TIMEOUT: 500` => `CONNECTION_TIMEOUT: 10000`
+    /// `LOG_FILE_VERBOSITY: info` => `LOG_FILE_VERBOSITY: debug`
```

Comment:
Reuse capture groups? Can you elaborate, I'm not following?

Reply:
like fs.mapping
```
    /// ### feature.fs.mapping {#feature-fs-mapping}
    ///
    /// Specify map of patterns that if matched will replace the path according to specification.
    ///
    /// *Capture groups are allowed.*
    ///
    /// Example:
    /// ```json
    /// {
    ///   "^/home/(?<user>\\S+)/dev/tomcat": "/etc/tomcat"
    ///   "^/home/(?<user>\\S+)/dev/config/(?<app>\\S+)": "/mnt/configs/${user}-$app"
    /// }
    /// ```
    /// Will do the next replacements for any io operaton
    ///
    /// `/home/johndoe/dev/tomcat/context.xml` => `/etc/tomcat/context.xml`
    /// `/home/johndoe/dev/config/api/app.conf` => `/mnt/configs/johndoe-api/app.conf`
    ///
    /// - Relative paths: this feature (currently) does not apply mappings to relative paths, e.g.
    ///   `../dev`.
 ```

Reply:
Hmm but doesn't `fancy_regex` also support capture groups? Like if the user has a config of (imagining these are env vars instead of paths):

```json
{
  "^/home/(?<user>\\S+)/dev/tomcat": "/etc/tomcat"
  "^/home/(?<user>\\S+)/dev/config/(?<app>\\S+)": "/mnt/configs/${user}-$app"
}
```

it should work the same?

Reply:
you use match, not replace - but in any case now that I think about it it's less relevant in this use case anyway.

Reply:
Ah I see.